components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/spark/transformations/DiscardedRowsFilter.java

```diff
@@ -41,11 +41,13 @@ public Boolean call(String[] tokens) throws Exception {
         try {
             Boolean keep = true;
             for (Integer index : indices) {
-                if (index >= tokens.length || MLConstants.EMPTY.equals(tokens[index]) || MLConstants.NA.equals(tokens[index])) {
+                if (index >= tokens.length || MLConstants.EMPTY.equals(tokens[index]) ||
```

Comment:
I tried to use this, but since it is required to match the values to strings such as `""`, `EnumUtils.isValidEnum()` method cannot be used. 
But a solution like below can be used:
```
public enum MISSING_VALUES {
        EMPTY(""), NA("NA"), QUESTION("?");

        private final String value;
        private MISSING_VALUES(final String str) {
            this.value = str;
        }

        @Override
        public String toString() {
            return value;
        }

        public static boolean contains(String s) {
            for (MISSING_VALUES val : values())
                if (val.toString().equals(s))
                    return true;
            return false;
        }
    }
```
Then something like the following can be used to check whether a given string is a missing value:
`MISSING_VALUES("?")`